hazelcast/src/test/java/com/hazelcast/cluster/ClusterShutdownTest.java

```diff
@@ -56,11 +57,19 @@ private void testClusterShutdownWithSingleMember(final ClusterState clusterState
         final Thread thread = new Thread(new Runnable() {
             @Override
             public void run() {
-                instances[0].getCluster().shutdown();
+                while (true) {
+                    try {
+                        instances[0].getCluster().shutdown();
+                        break;
+                    } catch (Exception ignored) {
```

Comment:
Ok. 

It still feels funny for the above reasons. In my book shutdowns should not fails because they can cause problems for other components in a system that also need to shut down. It can leave a system in an undefined half shut down state